pkg/controllers/provisioning/scheduling/inflightnode.go

```diff
@@ -87,9 +90,18 @@ func (n *InFlightNode) Add(pod *v1.Pod) error {
 		return err
 	}
 
-	if err := n.hostPortUsage.Add(pod); err != nil {
+	if err := n.hostPortUsage.Validate(pod); err != nil {
```

Comment:
Since we're taking a copy, doesn't it simplify things to collapse Add and Validate into the same method like we had before?

```if err := n.hostPortUsage.Add(pod); err != nil {`

Reply:
No, this introduces a bug that can cause us to create more nodes than necessary.  The pods are registered to host port usage, and then if a few lines later the pod is rejected for scheduling by requirements or instance type compatibility issues we would need to then remove it from the host port usage.  We could do that, but there's a few places and it's cleaner to compatibility check everything with respect to scheduling, and if it all passes, commit the changes.